Add keyboard shortcuts to titles

kommunarr opened the pull request:
# Add keyboard shortcuts to titles

<!-- Thanks for sending a pull request! Make sure to follow the contributing guidelines. -->
<!-- Important note, we may remove your pull request if you do not use this provided PR template correctly. -->

## Pull Request Type
<!-- Please select what type of pull request this is: [x] -->
- [ ] Bugfix
- [X] Feature Implementation
- [ ] Documentation
- [ ] Other

## Related issue
closes #1587

## Description
- Adds keyboard shortcut information to their button labels, e.g., `Mute (m)`
  - This includes the video player buttons, the feed refresh buttons, and the top nav HistoryForward/HistoryBackward/OpenNewWindow buttons, but NOT Electron context menu labels
- **Implementation note:** To do this properly for the built-in Shaka controls that we don't modify in our code, this PR updates the base Shaka label localization values to include the shortcut whenever the Shaka locale is set.
- **Implementation note:** Constants were added for the aforementioned keyboard shortcuts, including updating their existing code instances to use the constants.

## Screenshots <!-- If appropriate -->
<!-- Please add before and after screenshots if there is a visible change. -->
![Screenshot_20241013_181816](https://github.com/user-attachments/assets/4717569c-c5c8-470d-b1f2-b6cdd6bf7f3c)


## Testing <!-- for code that is not small enough to be easily understandable -->
- Test that buttons with corresponding keyboard shortcuts have the shortcut referenced in the label, be they in "on" or "off" mode (e.g., `Full screen (f)` / `Exit full screen (f)`). See [the official list of shortcuts](https://docs.freetubeapp.io/usage/keyboard-shortcuts/) for reference.

## Desktop
<!-- Please complete the following information-->
- **OS:** OpenSUSE
- **OS Version:** TW

## Additional context
The addition of the new constants should help make enhancements like #251 easier in the future.

PikachuEXE commented on src/constants.js:
```diff
@@ -113,6 +113,40 @@ const SyncEvents = {
+    FULLWINDOW: 's',
+    THEATRE_MODE: 't',
+    TAKE_SCREENSHOT: 'u',
+    VOLUME_UP: 'Up Arrow',
+    VOLUME_DOWN: 'Down Arrow',
+    SMALL_REWIND: 'Left Arrow',
+    SMALL_FAST_FORWARD: 'Right Arrow',
+    LAST_CHAPTER: 'Cmd/Ctrl+Left Arrow',
+    NEXT_CHAPTER: 'Cmd/Ctrl+Right Arrow',
```
Are these used?

kommunarr replied:
Nope, these are currently just here for future use. I'll mark that

PikachuEXE commented on src/constants.js:
```diff
@@ -113,6 +113,40 @@ const SyncEvents = {
 
+const KeyboardShortcuts = {
+  APP: {
+    HISTORY_BACKWARD: 'Alt+Left Arrow',
+    HISTORY_FORWARD: 'Alt+Right Arrow',
+    NEW_WINDOW: 'Cmd/Ctrl+N'
```
Maybe add a comment about these are used as labels and not in `case` only

PikachuEXE commented:
Label for captions
![image](https://github.com/user-attachments/assets/836e41a5-709f-4625-bfb6-d7f538873e2d)
![image](https://github.com/user-attachments/assets/82ff53a2-5766-4353-ac47-8e958e2bcf26)

kommunarr commented:
Another good catch; I wasn't testing with a video that had a captions option. Just fixed the issue and added logic preventing label modification if the Shaka localization key does not exist. Also note that this specific Captions label will be the only one I've seen with some text to the right of the shortcut. This is fine IMO and more work than desired to individually "fix".

![Screenshot_20241014_080659](https://github.com/user-attachments/assets/d8ccd442-fc1c-4d82-836e-278e820bbf72)

efb4f5ff-1298-471a-8973-3d47447115dc reviewed (commented):
Pressing `c` doesnt enable/disable captions

kommunarr commented:
Added localization of shortcuts & multi-key shortcuts as well as a dangling instance of the wrong constant name for `CAPTIONS`

efb4f5ff-1298-471a-8973-3d47447115dc commented:
Question: History tab and Settings tab also got shortcuts. Is there a reason those weren't added?

kommunarr commented:
@efb4f5ff-1298-471a-8973-3d47447115dc I specifically added the shortcuts with corresponding button labels. The closest equivalent to those would be the corresponding side nav options, which I suppose would work. I can add those as well.

PikachuEXE reviewed (dismissed):
Tested video player, sidebar, top nav button labels, refresh button

Only on MacOS though

kommunarr commented:
Still on the fence about whether we should use the Mac key symbols for `darwin` (⌘/⇧/⌥/etc.)

efb4f5ff-1298-471a-8973-3d47447115dc commented:
> Still on the fence about whether we should use the Mac key symbols for `darwin` (⌘/⇧/⌥/etc.)
> 
> Edit: did it. I used the version of the arrow icons that are used in Mac context menus, but it is worth noting that FF for Mac uses the more standard-looking ←→↑↓. I'm also open to using the arrow icons for non-Mac systems; I just went with the text version of those from looking at FF for Linux.

@PikachuEXE what do you think of the change in the latest commit

PikachuEXE commented:
I think there is no need to use short form on labels
The places I see short forms are app menu bar & keymap menus with limited spaces
Even short forms should be used the arrows chars should be ←→↑↓ at least
I haven't checked how this looks on Windows yet

Zen Browser
![image](https://github.com/user-attachments/assets/fa3081db-dd62-44c4-81a2-2b1a5557932d)
RubyMine
![image](https://github.com/user-attachments/assets/d79486d0-91d9-48ad-9ef9-a5f66ce2e607)
Vivaldi
![image](https://github.com/user-attachments/assets/dd2a84b4-29a2-4a07-bf09-b43af16c723f)

kommunarr commented:
I can change the arrow symbols. I'm not sure if we have other active contributors who are Mac users to provide additional opinions, but maybe if any users have any thoughts

kommunarr commented:
Updated the arrow icons for Mac:

<img width="153" alt="Screenshot 2024-10-23 at 8 24 31 AM" src="https://github.com/user-attachments/assets/f75ba58f-5371-4959-a29e-c33832dca4ca">

absidue commented:
I preferred the stats button before this pull request a lot more as it matches the appearance of the other controls, now it looks out of place in the player.

Before:

![before](https://github.com/user-attachments/assets/1e6efe26-3b69-488b-850c-e1efe4785258)

After:
![after](https://github.com/user-attachments/assets/0cb56fc4-c427-490e-83a7-adc4623ada07)

Other controls:
![other-controls](https://github.com/user-attachments/assets/977915cd-9deb-48b4-984f-e9d9202b4a15)

kommunarr commented:
Ah, for the stats button, you're referring to the text content? I noticed that the labels we were setting for it weren't being used, so I thought that was an error. I can change that back if desired.

I can also look into removing the shortcuts from the overflow menu if that is desired as well. My only concern with that is that I'm not sure if we will be adding more shortcuts or moving more controls to a default overflow menu in the future.

efb4f5ff-1298-471a-8973-3d47447115dc reviewed (changes requested):
Fullwinodw doesnt show shortcut key in overflow menu

kommunarr commented:
Sorry, I'm still a bit confused on what is being asked to be changed. Just fixing full window to have the shortcut show in the overflow menu? @absidue @efb4f5ff-1298-471a-8973-3d47447115dc

efb4f5ff-1298-471a-8973-3d47447115dc commented:
> Just fixing full window to have the shortcut show in the overflow menu?

correct

PikachuEXE commented on src/renderer/components/ft-shaka-video-player/ft-shaka-video-player.js:
```diff
@@ -39,6 +40,19 @@ const RequestType = shaka.net.NetworkingEngine.RequestType
+  PLAY: KeyboardShortcuts.VIDEO_PLAYER.PLAYBACK.PLAY,
+  PAUSE: KeyboardShortcuts.VIDEO_PLAYER.PLAYBACK.PLAY,
+  PICTURE_IN_PICTURE: KeyboardShortcuts.VIDEO_PLAYER.GENERAL.PICTURE_IN_PICTURE,
+  ENTER_PICTURE_IN_PICTURE: KeyboardShortcuts.VIDEO_PLAYER.GENERAL.PICTURE_IN_PICTURE,
+  EXIT_PICTURE_IN_PICTURE: KeyboardShortcuts.VIDEO_PLAYER.GENERAL.PICTURE_IN_PICTURE,
```
These 2 might be used but really difficult to understand
Might need a comment to explain their existence

kommunarr replied:
Added a comment now

ChunkyProgrammer commented on src/renderer/views/Popular/Popular.js:
```diff
@@ -0,0 +1,108 @@
+import { defineComponent } from 'vue'
```
This file no longer exists in dev and was migrated to Popular.vue (composition api)

kommunarr replied:
Good catch, I missed this on resolving the merge conflict. Updated now
